### PR TITLE
security(cdc): fix some security problems

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

fix https://github.com/pingcap/ticdc/issues/3696

preview the fixed results [here](https://ci.pingcap.net/blue/organizations/jenkins/debug-cdc-verify-ci/detail/debug-cdc-verify-ci/6/pipeline)

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡 @hi-rustin  @asddongmen

File: cdc/sink/producer/kafka/config.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (c *Config) CompleteByOpts(sinkURI *url.URL, replicaConfig *config.ReplicaC
 	params := sinkURI.Query()
 	s := params.Get("partition-num")
 	if s != "" {
-		a, err := strconv.Atoi(s)
```
Comment (lance6716) [inline cdc/sink/producer/kafka/config.go:75]: why this is a security problem :joy:

Comment (Ehco1996) [reply cdc/sink/producer/kafka/config.go:75]: becasue  in L79 `c.PartitionNum = int32(a)` try convert int to int32 😂

File: tests/integration_tests/bank/case.go [hunk 2]
```diff
@@ -619,11 +619,10 @@ func getDownStreamSyncedEndTs(ctx context.Context, db *sql.DB, tableName string)
 }
 
 func tryGetEndTs(db *sql.DB, tableName string) (result string, ok bool) {
-	query := fmt.Sprintf("admin show ddl jobs where table_name = '%s'", tableName)
+	query := "admin show ddl jobs where table_name = ?"
 	log.Info("try get end ts", zap.String("query", query))
-
 	var line dataRow
-	row := db.QueryRow(query)
+	row := db.QueryRow(query, tableName)
```
Comment (okJiang) [inline tests/integration_tests/bank/case.go:625]: What's the difference in this change?

Comment (Ehco1996) [reply tests/integration_tests/bank/case.go:625]: the old one may casue sql jnjection see more [here](https://ci.pingcap.net/blue/organizations/jenkins/atom-gosec/detail/atom-gosec/840/tests/)

Comment (okJiang) [reply tests/integration_tests/bank/case.go:625]: see details https://cwe.mitre.org/data/definitions/89.html

**Review by okJiang — APPROVED** [review]

File: cdc/sink/producer/kafka/config.go [hunk 3]
```diff
@@ -84,7 +84,7 @@ func (c *Config) CompleteByOpts(sinkURI *url.URL, replicaConfig *config.ReplicaC
 
 	s = params.Get("replication-factor")
 	if s != "" {
-		a, err := strconv.Atoi(s)
+		a, err := strconv.ParseInt(s, 10, 16)
```
Comment (hi-rustin) [inline cdc/sink/producer/kafka/config.go:87]: Why is this 16?

File: cdc/sorter/unified/file_backend.go [hunk 4]
```diff
@@ -65,7 +65,7 @@ func newFileBackEnd(fileName string, serde encoding.SerializerDeserializer) (*fi
 }
 
 func (f *fileBackEnd) reader() (backEndReader, error) {
-	fd, err := os.OpenFile(f.fileName, os.O_RDWR, 0o644)
+	fd, err := os.OpenFile(f.fileName, os.O_RDWR, 0o600)
```
Comment (hi-rustin) [inline cdc/sorter/unified/file_backend.go:68]: /cc @liuzix 

Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

File: cdc/sink/producer/kafka/config.go (hunk 3 above)
Comment (Ehco1996) [reply cdc/sink/producer/kafka/config.go:87]: because in L91 `	c.ReplicationFactor = int16(a)`

**Review by asddongmen — APPROVED** [review]

**Pull request merged**